webidls/Browser.webidl

```diff
@@ -0,0 +1,115 @@
+/**
+  * Browser is the equivalent of a top level Constellation::Frame.
+  * Browser is a top level browsing context.
+  * ~= Tab, ~= Webview
+  */
+
+[Constructor(DOMString contextName, StorageSession storageSession)]
+interface Browser {
+  // FIXME: how is this attached to a viewport? For swapping, it's important to be able to attach and detach
+
+  Promise<void> setHandler(BrowserHandler handler);
+
+  // Used to expose special JS APIs.
+  // For example, can be used to expose the Browser API or Web Extensions APIs for content
+  // See: https://developer.chrome.com/extensions/content_scripts#execution-environment
+  Promise<void> addJSModuleResolver(/*FIXME*/, boolean onlyForFrameScript);
+
+  void destroy();
+
+  // Can't change after browser creation
+  readonly attribute DOMString browsingContextName;
+  readonly attribute StorageSession storageSession;
+
+  // Will fail if entries already exist.
+  // This can be used for session restore, or to undo tab-close.
+  // On success, historyEntries is filled, active entry's pipeline is NOT pending.
+  // If necessary, other pipelines can be restored via HistoryEntry.restorePipeline()
+  Promise<void> restoreEntries(FrozenArray<LoadData> data, unsigned long activeIndex);
+
+  readonly attribute Sequence<HistoryEntry> historyEntries;
+  readonly attribute unsigned long activeEntryIndex;
+  Promise<void> setActiveEntryIndex(unsigned long);
+
+  // Use to load a new URL. will create a new pipeline and navigate to the
+  // pipeline once not pending
+  Promise<HistoryEntry> navigate(LoadData loadData, optional Pipeline opener);
+
+  // Will fail if Browser.session != pipeline.session.
+  // Useful with prerendering pipelines
+  Promise<Pipeline> createPrerenderingPipeline(LoadData loadData);
+  Promise<HistoryEntry> navigateToPipeline(Pipeline pipeline, optional Pipeline opener);
+
+
+  // FIXME: do we want to give access to autopurge and maxLivePipeline?
+  // Purging means killing the pipeline. The entry stays intact
+  readonly attribute boolean doPipelinesAutopurge;
+  // Maximum number of live pipeline before and after the current entry
+  Promise<void> enableAutoPurgePipelines(unsigned long before, unsigned long after);
+  Promise<void> disableAutoPurgePipelines();
+
+  // Popup blocker, tracking content blocker, mixed content blocker,
+  // and safari-like content blocker. See ContentBlockers.webidl
+  // FIXME: not cool. We can't have multiple custom blockers.
+  readonly attribute Sequence<ContentBlockerDescription> contentBlockers;
+  Promise<void> setContentBlockers(Sequence<ContentBlockerDescription>);
```

Comment:
I am not entirely sure if there are use cases for mutating `contentBlockers` list at runtime. Unless there is i would also just make it part of constructor config.  In fact if mutation of `contentBlockers` is rare it still would make more sense to make it part of constructor config instead & such case could be covered by creating new `Browser` with same entries but different `contentBlockers`.